src/shipchain_common/test_utils/httpretty_asserter.py

```diff
@@ -14,57 +14,80 @@
 limitations under the License.
 """
 import pytest
-from httpretty import HTTPretty
+import responses
 
 from urllib.parse import urlparse
-from ..utils import parse_urlencoded_data
+from ..utils import parse_urlencoded_data, parse_value
 
 
-class HTTPrettyAsserter(HTTPretty):
-    @classmethod
-    def _parse_calls_into_list(cls):
+class ResponsesHTTPrettyWrapper:
+    def __init__(self):
+        self.mock = responses.mock
+
+    def __getattr__(self, item):
+        return getattr(self.mock, item)
+
+    def register_uri(self,  # noqa
+                     method,
+                     uri,
+                     body="HTTPretty :)",
+                     adding_headers=None,
+                     status=200,
+                     match_querystring=False,
+                     **headers
+                     ):
+        try:
+            self.mock.replace(method_or_response=method, url=uri, body=body, status=status,
+                              match_querystring=match_querystring, headers=headers)
+        except ValueError:
+            self.mock.add(method=method, url=uri, body=body, status=status,
+                          adding_headers=adding_headers, match_querystring=match_querystring, headers=headers)
+
+    @property
+    def latest_requests(self):
+        return self.mock.calls
+
+    def reset(self):
+        self.mock._calls.reset()  # pylint:disable=protected-access
+
+class ResponsesAsserter(ResponsesHTTPrettyWrapper):
+    def _parse_calls_into_list(self):
         calls_list = []
-        assert cls.latest_requests, 'Error: No calls made to be parsed.'
-        for index, call in enumerate(cls.latest_requests):
-            if call is None:
-                # If calling cls.reset(), latest_requests are unable to build.
-                # Setting the current call to None and iterating through this way ensures that only calls from the
-                # individual test are set in the call_list.
-                continue
-
-            url = urlparse(call.path)
-            if call.headers.get('content-type', '') in ('application/json', 'text/json'):
-                body = call.parsed_body
+        assert self.latest_requests, 'Error: No calls made to be parsed.'
```

Comment:
maybe it would make sense to remove this assertion, and instead include an assert that there are/aren't calls to parse to ensure no requests were made in a function.